### DISCUSSION TITLE
Using cyclonedx-editor-validator in python script

### DISCUSSION THREAD

**Discussion by cedricwritescode** [Festo-se/cyclonedx-editor-validator#133] [Q&A] [answered]
I want to use cdx-ev from a python script.
Besides running CLI commands using `subprocess.run`, is it intended to import this package (i.e. `from cyclonedx-editor-validator import *`) and calling methods directly? If so, which methods would correspond to the CLI commands?

**Comment by italvi** [comment]
@cedricwritescode I think the best starting point is to look at our `__main__.py`, this should answer most of your questions, more precisely the `import` (the module is called `cdxev`, not `cyclonedx-editor-validator`) and `invoke_{command}` there.

**Reply by cedricwritescode** [reply]
Currently, cdx-ev is intended to be a command-line tool. Are there any (long term) plans to also officially support its use in python directly as mentioned above? If not, I think going with the official usage (command-line, `subprocess.run` with python) would be more appropriate in my case.

**Reply by italvi** [reply, marked as answer]
Currently no official support for the use within scripts is planned. The primarily use-case for the tool is the usage within pipelines, so I would suggest to stick to your approach, using `subprocess.run`.
